tests/Doctrine/Tests/DBAL/Functional/TransactionTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Functional;
+
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\Tests\DbalFunctionalTestCase;
+use Throwable;
+use function in_array;
+use function sleep;
+
+class TransactionTest extends DbalFunctionalTestCase
+{
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        if (in_array($this->connection->getDatabasePlatform()->getName(), ['mysql'])) {
+            return;
+        }
+
+        $this->markTestSkipped('Restricted to MySQL.');
+    }
+
+    protected function tearDown() : void
+    {
+        $this->resetSharedConn();
+
+        parent::tearDown();
+    }
+
+    public function testCommitFalse() : void
+    {
+        $this->connection->query('SET SESSION wait_timeout=1');
+
+        $table = new Table('foo');
+        $table->addColumn('id', 'integer');
+        $table->setPrimaryKey(['id']);
+
+        $this->connection->getSchemaManager()->createTable($table);
+
+        self::assertEquals('foo', $table->getName());
+        try {
+            $this->assertTrue($this->connection->beginTransaction());
+            $this->connection->executeUpdate('INSERT INTO foo (id) VALUES(1)');
+
+            sleep(2); // during the sleep mysql will close the connection
+
+            $this->assertFalse(@$this->connection->commit()); // we will ignore `MySQL server has gone away` warnings
```

Comment:
This is the test I was asking for!

Reply:
Please disable handling warnings explicitly: `PHPUnit\Framework\Error\Warning::$enabled = false`. Then enable back on teardown.

Reply:
I have used `Warning::$enabled = false` and removed `@`. It [does not throw exceptions, but the warnings mark the tests as failed](https://travis-ci.org/doctrine/dbal/builds/550168914).
Did I miss something?

Reply:
This is weird. It works for me in my local environment on c08baa3e5095260049f6e39bbe6a75ae25d39cef without any code modifications (on both `mysqli` and `pdo_sqlsrv`).

It seems to mark the test risky because it produces unexpected output as a side effect of reporting the warning. If you can't reproduce this failure locally and fix it appropriately, please feel free to revert to `@`.

Reply:
Unfortunately I can't solve this without `@`.